java/arcs/android/common/resurrection/DbHelper.kt

```diff
@@ -48,7 +48,16 @@ class DbHelper(
         db.transaction { CREATE.forEach(db::execSQL) }
     }
 
-    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) = Unit
+    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
+        if (oldVersion == RESURRECTION_DB_OLD_VERSION && newVersion == RESURRECTION_DB_VERSION) {
```

Comment:
RESURRECTION_DB_OLD_VERSION is a poor name for this, I think.. since versions can be skipped.

Something along this pattern has been useful to me in the past:

* Place the migration SQL statements in the companion object, under a name like "VERSION_2_MIGRATION"
* Use a more general check like: `if (2 in (oldVersion + 1)..newVersion)`
* Within the check, do the `execSQL` calls.

```kotlin
override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) = db.transaction {
    val upgradeRange = (oldVersion + 1)..newVersion
    if (2 in upgradeRange) {
        VERSION_2_MIGRATION.forEach(db::execSQL)
    }
}

....

companion object {
    const val RESURRECTION_DB_NAME = "..."
    const val RESURRECTION_DB_VERSION = 2

    private val CREATE = """
        // ....
    """.trimIndent()

    private val VERSION_2_MIGRATION = arrayOf(
        "DROP TABLE resurrection_requests",
        "DROP TABLE requested_notifiers",
        "DROP TABLE notifiers_by_component"
    ) + CREATE
}
```

Then, later on when there are more migrations to look for, `onUpgrade` can be as simple as adding more `if`s. E.g.

```kotlin
override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) = db.transaction {
    val upgradeRange = (oldVersion + 1)..newVersion
    if (2 in upgradeRange) {
        VERSION_2_MIGRATION.forEach(db::execSQL)
    }
    if (3 in upgradeRange) {
        VERSION_3_MIGRATION.forEach(db::execSQL)
    }
    if (4 in upgradeRange) {
        VERSION_4_MIGRATION.forEach(db::execSQL)
    }
    if (5 in upgradeRange) {
        VERSION_5_MIGRATION.forEach(db::execSQL)
    }
}
```

It could be tidied up even more with some kind of migration step array.. but that's starting to get into diminishing returns.